graphql/ui_plugin_build_baron.go

```diff
@@ -192,6 +192,11 @@ func GetSearchReturnInfo(taskId string, exec string) (*thirdparty.SearchReturnIn
 	buildBaronProjects := BbGetConfig(settings)
 	bbProj, ok := buildBaronProjects[t.Project]
 
+	// if the project uses custom ticket creation, don't use the default search
+	if bbProj.TaskAnnotationSettings.FileTicketWebHook.Endpoint != "" {
+		return nil, bbConfig, errors.Errorf("Build Baron project for %s is configured to use a custom webhook", t.Project)
+	}
+
 	if !ok {
 		return nil, bbConfig, errors.Errorf("Build Baron project for %s not found", t.Project)
```

Comment:
I'm looking at where GetSearchReturnInfo is used (sorry if I reviewed this the first time and missed it): in the BuildBaron query resolver, you first check if bbConfig has ProjectFound or SearchConfigured set to false, in which case you ignore the error. But because bbConfig isn't set to anything at this point, that means you'll always ignore the error. That seems wrong.

Reply:
I think that's still fine for that resolver, because if the build baron is not configured, then there are no errors that we care about over there. We don't want to error on a build baron error for a team that doesn't even know what the build baron is and doesn't interact with it.

Reply:
In neither places this is called do we look at the error though. There's no point in returning an error if it's not something we consider an error, and the caller of the function shouldn't have to know what kind of errors do or do not matter. Say the project _does_ have the build baron configured: then the error "Error searching for tickets" would be a relevant error, for example. We shouldn't be returning errors we don't want to know about.

Reply:
Is the argument that we SHOULD be checking the error or that we shouldn't be returning it?

Reply:
We _should_ always check errors, so if there are errors we don't care about, we shouldn't return them. (We could grip.Debug them or something)